src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,714 @@
+        shares = _deposit(assets, msg.sender);
+    }
+
+    /// @notice deposit liquidity assets in exchange for liquidity provider (LP) shares.
+    /// This function is restricted to the Zap contract.
+    /// @param assets The amount of liquidity assets to deposit
+    /// @param liquidityProvider The address of the liquidity provider
+    /// @return shares The amount of shares that were minted
+    function deposit(uint256 assets, address liquidityProvider) external returns (uint256 shares) {
+        require(msg.sender == zap, "Only Zap");
+
+        shares = _deposit(assets, liquidityProvider);
+    }
+
+    /// @dev Internal logic for depositing liquidity assets in exchange for liquidity provider (LP) shares.
+    function _deposit(uint256 assets, address liquidityProvider) internal returns (uint256 shares) {
+        // Calculate the amount of shares to mint after the performance fees have been accrued
+        // which reduces the available assets, and before new assets are deposited.
+        shares = convertToShares(assets);
+
+        // Transfer the liquidity asset from the sender to this contract
+        IERC20(liquidityAsset).transferFrom(msg.sender, address(this), assets);
+
+        // mint shares
+        _mint(liquidityProvider, shares);
+
+        // Add the deposited assets to the last available assets
+        lastAvailableAssets += SafeCast.toInt128(SafeCast.toInt256(assets));
+
+        // Check the liquidity provider caps after the new assets have been deposited
+        if (capManager != address(0)) {
+            ICapManager(capManager).postDepositHook(liquidityProvider, assets);
+        }
+
+        emit Deposit(liquidityProvider, assets, shares);
+    }
+
+    /// @notice Preview the amount of assets that would be received for burning a given amount of shares
+    /// @param shares The amount of shares to burn
+    /// @return assets The amount of liquidity assets that would be received
+    function previewRedeem(uint256 shares) external view returns (uint256 assets) {
+        assets = convertToAssets(shares);
+    }
+
+    /// @notice Request to redeem liquidity provider shares for liquidity assets
+    /// @param shares The amount of shares the redeemer wants to burn for liquidity assets
+    /// @return requestId The index of the withdrawal request
+    /// @return assets The amount of liquidity assets that will be claimable by the redeemer
+    function requestRedeem(uint256 shares) external returns (uint256 requestId, uint256 assets) {
+        // Calculate the amount of assets to transfer to the redeemer
+        assets = convertToAssets(shares);
+
+        requestId = nextWithdrawalIndex;
+        uint120 queued = SafeCast.toUint120(withdrawsQueued + assets);
+        uint40 claimTimestamp = uint40(block.timestamp + claimDelay);
+
+        // Store the next withdrawal request
+        nextWithdrawalIndex = SafeCast.toUint16(requestId + 1);
```

Comment:
⚪️Given that we aren't structing together variables to pack into a single write, I think it's better to move this increment up to right below the `requestId` read. This way it's read, write, done for this var.

Reply:
Fix proposed in this commit [9b74872](https://github.com/OriginProtocol/arm-oeth/pull/33/commits/9b74872bc765c3a0eb9a1e6e2524572410a49a89).